kasa/modules/emeter.py

```diff
@@ -43,31 +43,48 @@ async def get_realtime(self):
         """Return real-time statistics."""
         return await self.call("get_realtime")
 
-    async def get_daystat(self, *, year, month, kwh=True):
-        """Return daily stats for the given year & month."""
-        raw_data = await super().get_daystat(year=year, month=month)
-        return self._emeter_convert_emeter_data(raw_data["day_list"], kwh)
+    async def get_daystat(self, *, year=None, month=None, kwh=True) -> Dict:
+        """Return daily stats for the given year & month as a dictionary of {day: energy, ...}."""
+        data = await self.get_raw_daystat(year=year, month=month)
+        data = self._convert_stat_data(data["day_list"], entry_key="day", kwh=kwh)
+        return data
+
+    async def get_monthstat(self, *, year=None, kwh=True) -> Dict:
+        """Return monthly stats for the given year as a dictionary of {month: energy, ...}."""
+        data = await self.get_raw_monthstat(year=year)
+        data = self._convert_stat_data(data["month_list"], entry_key="month", kwh=kwh)
+        return data
 
-    async def get_monthstat(self, *, year, kwh=True):
-        """Return monthly stats for the given year."""
-        raw_data = await super().get_monthstat(year=year)
-        return self._emeter_convert_emeter_data(raw_data["month_list"], kwh)
+    def _convert_stat_data(self, data, entry_key, kwh=True) -> Dict:
+        """Return emeter information keyed with the day/month.
 
-    def _emeter_convert_emeter_data(self, data, kwh=True) -> Dict:
-        """Return emeter information keyed with the day/month.."""
-        response = [EmeterStatus(**x) for x in data]
+        The incoming data is a list of dictionaries:
+            [{'year':      int,
```

Comment:
```suggestion
        The incoming data is a list of dictionaries::

            [{'year':      int,
```
See my comment wrt. documentation formatting below.